managed/resource_cluster.go

```diff
@@ -1044,6 +1145,16 @@ func (r resourceCluster) Update(ctx context.Context, req tfsdk.UpdateResourceReq
 		}
 	}
 
+	// Pause the cluster if the desired state is set to 'Paused' and it is active currently
+	if state.DesiredState.Value == "Active" && (!plan.DesiredState.Unknown && plan.DesiredState.Value == "Paused") {
```

Comment:
Please add a UT for these two transitions (line 1014 and 1148) where we trigger the switch in your next PR (for 1014, we should have both as not specified and specified as Active).